### PR TITLE
Feat 10316: Can now do a dryRun of syncIndexes using Model.diffIndexes()

### PR DESCRIPTION
**Pull request by IslandRhythms**

### CODE REVIEW

File: test/model.indexes.test.js [hunk 1]
```diff
@@ -683,5 +683,17 @@ describe('model', function() {
         yield User.collection.drop();
       });
     });
+    it('should do a dryRun feat-10316', function() {
+      return co(function*() {
+        const userSchema = new mongoose.Schema({ username: String }, { password: String }, { email: String });
+        userSchema.index({ password: 1 });
+        userSchema.index({ email: 1 });
+        const User = db.model('Upson', userSchema);
```
Comment (vkarpov15) [inline test/model.indexes.test.js:691]: Add a `yield User.init()` here to wait for all indexes to finish building

File: lib/model.js [hunk 2]
```diff
@@ -1411,6 +1411,42 @@ Model.syncIndexes = function syncIndexes(options, callback) {
...
+          continue;
+        }
+
+        for (const schemaIndex of schemaIndexes) {
+          const key = schemaIndex[0];
+          const options = _decorateDiscriminatorIndexOptions(this,
+            utils.clone(schemaIndex[1]));
+          if (isIndexEqual(key, options, index)) {
+            found = true;
+          } else {
```
Comment (vkarpov15) [inline lib/model.js:1436]: You may end up adding the same index to `toCreate` multiple times. If you have 3 indexes in MongoDB, you'll execute the `schemaIndexes()` loop 3 times and end up adding the same schema index twice each time.

You would likely need to write a separate `for` loop to correctly compute `toCreate`

**Review by vkarpov15 — CHANGES_REQUESTED** [review]

**Comment by IslandRhythms** [comment]
Do you want me to go back in and add the comments to describe what diffIndexes() does or do you got it?

**Comment by vkarpov15** [comment]
@IslandRhythms please add comments, that would be very helpful :+1:

**Review by vkarpov15 — APPROVED** [review]
Good work, this looks great :+1:

**Pull request merged**